### PR TITLE
test: install Cilium before creating any pods in upgrade test

### PR DESCRIPTION
**Pull request by aanm**
As Cilium is installed with with the clean state it can delete the
connectivity endpoints recently created before deploying Cilium causing
the test to fail.

Fixes: fde53a7 ("test: add etcd-operator to CI")
Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5850)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: test/k8sT/Updates.go [hunk 1]
```diff
@@ -144,6 +138,13 @@ func InstallAndValidateCiliumUpgrades(kubectl *helpers.Kubectl, oldVersion, newV
...
 		ExpectCiliumReady(kubectl)
 
+		By("Installing kube-dns")
+		_ = kubectl.Apply(helpers.DNSDeployment())
+
+		// Deploy the etcd operator
+		By("Deploying etcd-operator")
+		err = kubectl.DeployETCDOperator()
+		Expect(err).To(BeNil(), "Unable to deploy etcd operator")
 		By("Cilium %q is installed and running", oldVersion)
```
Comment (eloycoto) [inline test/k8sT/Updates.go:150]: Quick one, with that changes that line should be in the top, 140 no? After ExpectCiliumReady I think.

I'm refering to this line: 
```
By("Cilium %q is installed and running", oldVersion)
```

**Review by eloycoto — CHANGES_REQUESTED** [review]

File: test/k8sT/Updates.go (hunk 1 above)
Comment (aanm) [reply test/k8sT/Updates.go:150]: You know, when I looked at this line before pushing the branch it seemed to "close" to the line 147 I read the `"Cilium %q is installed and running"` and it seemed that was in the right place. 

Fixed

**Comment by aanm** [comment]
test-me-please

**Review by eloycoto — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

EDIT: test failure is unrelated

**Review by eloycoto — APPROVED** [review]

**Pull request merged**